src/lib.rs

```diff
@@ -674,8 +674,15 @@ pub mod sync {
         ///
         /// Returns `None` if the cell is empty.
         pub fn get_mut(&mut self) -> Option<&mut T> {
-            // Safe b/c we have a unique access.
-            unsafe { &mut *self.0.value.get() }.as_mut()
+            if self.0.is_initialized() {
+                // Safe b/c we have a unique access.
+                unsafe {
+                    let slot: &mut MaybeUninit<T> = &mut *self.0.value.get();
+                    Some(&mut *slot.as_mut_ptr())
+                }
```

Comment:
It's safe not only because of the uniqueness of the access, but also b/c of the initialized flag. Not sure if it's obvious enough to not mention here.